### DISCUSSION TITLE
IE 11 css overrides?

### DISCUSSION THREAD

**Discussion by bkyam** [vercel/next.js#12071] [Help] [answered]
I'm rather new to react and nextjs. We are building out a new site using nextjs 9.3+ and I'm loving the module.scss approach.

I'm having a difficult time figuring out how to manage very specific IE11 css overrides. 

I understand that there are now a lot of great css polyfills for IE11 and such, but they don't address all situations.

How can I effectively target IE11 in my module.scss files, or is there a better way to handle it?

Thanks.

**Comment by jamesmosier** [comment, marked as answer]
Could you perhaps provide an example (or scenario) of what you are looking for?

**Reply by bkyam** [reply]
Thanks for reaching out!

Say I have a component, Button.js

```
import styles from './Button.module.scss'
class Button extends React.component {
...
render() {
return (
<div className{styes.someButtonStyle}>...</div>
)
}}
export default Button
```

As you know, the styles from Button.module.scss take precedence above everything else. I'd like to be able to add IE11 specific styles to the Button.module.scss file. I'm not finding a way that I can reliably override the styles in Button.module.scss.

Any ideas on how to accomplish that?

Alternatively, would it be possible to have a second scss file, ButtonIE.module.scss, that is only delivered if the browser is IE11? That'd keep that junk CSS out of production for non-IE11 browsers.

Sadly we have to support IE 11 :/

I appreciate any advice you have.

**Reply by jamesmosier** [reply]
I don't know of anyway to conditionally serve a stylesheet based on browser without a ton of leg work.

Personally, I believe that any CSS specific styles for IE11 would have such a minor effect on performance/file size that there are many other places that would be better served to focus on performance. Therefore I would just include the IE11 styles within your CSS modules and target them using something [like this](https://stackoverflow.com/a/22085269/1554860) (`@media all and (-ms-high-contrast:none) {}`). 

Sorry that probably isn't super helpful, but just my 2 cents 😄

**Reply by bkyam** [reply]
I appreciate it! I'll give it a try tomorrow.  thx!

**Reply by bkyam** [reply]
I just wanted to chime back in and let you know that approach turned out to be the easiest and most efficient. Thank you!

**Reply by jamesmosier** [reply]
awesome! glad to hear it 🙏

**Comment by ebs-PriyankaKanwar** [comment]
we have updated next JS, all things are working fine except one the CSS is not loaded in proper order in production mode, however, it's working fine in dev mode.

Here are the versions
Node: v13.7.0
React:^16.12.0
NextJs:9.1.6
OS: Linux - ubuntu

Here is what I got when did build:

├ static/pages/_app.js 16.1 kB
├ chunks/1899c3d08f1d4a9ab6e4e22050e66cd1600464e8.b9ea19.js 3.53 kB
├ chunks/48183758.ede116.js 65 B
├ chunks/57a5805a715b88beb8aec3ac43d8f962ef8879cc.53c4e4.js 2.34 kB
├ chunks/9ed67fcaced240be487e7a2880bfff868999eacb.76e83a.js 8.29 kB
├ chunks/ab68c4ca07cacf4e3048ee2d6f69810106b4b489.68a8b3.js 10.4 kB
├ chunks/d32b6132cd0996134c9f87f7274f553f791cd563.18dde5.js 6.39 kB
├ chunks/d6e4d54ae3ad4e326943499c7720eb830c61721b.b7693f.js 4.41 kB
├ chunks/framework.4dd100.js 40.3 kB
├ runtime/main.d3f0e8.js 13.2 kB
├ runtime/webpack.361a6a.js 1.59 kB
├ css/16c4518f007189356065.css 54.9 kB
└ css/f93669f8200e9712e425.css 30.6 kB

Here is my next.config :
const withFonts = require('next-fonts');
const withImages = require('next-images');
// const withCSS = require('@zeit/next-css')

const withPlugins = require('next-compose-plugins');
const nextEnv = require('next-env');
const dotenvLoad = require('dotenv-load');

dotenvLoad();

withFonts({
webpack(config, options) {
// config.plugins.push(new webpack.EnvironmentPlugin(localEnv))
return config;
}
})

module.exports = withPlugins([

nextEnv({
staticPrefix: 'CUSTOM_STATIC_',
publicPrefix: 'CUSTOM_PUBLIC_',
}),withFonts, withImages
]);

I have two things here
1 - why we are having two files of CSS when created build? How can we merge into one CSS file?
2 - How I can fix it so that it will work in the same manner as in dev mode?

Any suggestions will be helpful.
